core/src/raw/http_util/client.rs

```diff
@@ -78,6 +81,34 @@ impl HttpClient {
 
     /// Fetch a request in async way.
     pub async fn fetch(&self, req: Request<Buffer>) -> Result<Response<HttpBody>> {
+        self.fetcher.fetch(req).await
+    }
+}
+
+#[async_trait::async_trait]
+pub trait HttpFetch: Send + Sync {
```

Comment:
Please also add `Unpin + 'static` here, we don't allow users to implement `HttpFetch` over ref.